autoreject/tests/test_autoreject.py

```diff
@@ -318,3 +319,33 @@ def test_io():
     assert_array_equal(epochs_clean1.get_data(), epochs_clean3.get_data())
     assert_array_equal(reject_log1.labels, reject_log2.labels)
     assert_array_equal(reject_log1.labels, reject_log3.labels)
+
+
+def test_fnirs():
+    # Test that autoreject runs on fNIRS data.
+    raw = mne.io.read_raw_nirx(os.path.join(
+        mne.datasets.fnirs_motor.data_path(), 'Participant-1'))
+    raw.crop(tmax=1200)
+    raw = mne.preprocessing.nirs.optical_density(raw)
+    raw = mne.preprocessing.nirs.beer_lambert_law(raw)
+    events, _ = mne.events_from_annotations(raw, event_id={'1.0': 1,
+                                                           '2.0': 2,
+                                                           '3.0': 3})
+    event_dict = {'Control': 1, 'Tapping/Left': 2, 'Tapping/Right': 3}
+    epochs = mne.Epochs(raw, events, event_id=event_dict,
+                        tmin=-5, tmax=15,
+                        proj=True, baseline=(None, 0), preload=True,
+                        detrend=None, verbose=True)
+    # Test autoreject
+    ar = AutoReject()
+    assert len(epochs) == 37
+    epochs_clean = ar.fit_transform(epochs)
```

Comment:
sorry for my ignorance, how does interpolation work for this kind of data? Are there many channels?

Reply:
Currently in MNE we are replacing the data with the closest channel. Until I get around to developing something smarter, but that's not on my schedule for a few months. So maybe we should remove this?

The number of channels varies greatly. There are single channel systems through to high density fNIRS which probably has 100s of channels. The high density systems are commonly available though. For reference, my two labs typically use around 50 channels.

Reply:
Here is a figure of what a low (left) and high density (right) system would look like targeting the frontal cortex. This is from a commercial system.

green lines represent channels

![image](https://user-images.githubusercontent.com/748691/124337890-42877000-dbe8-11eb-9055-6b652d90fc1e.png)

Reply:
I see, got it. It's fine as such I think. Feel free to make follow up pull requests to improve the functionality. I'm guessing you look for the nearest channel of the same channel type?

Reply:
> I'm guessing you look for the nearest channel of the same channel type?

Correct